go/signedexchange/README.md

```diff
@@ -1,3 +1,83 @@
+    ```
+    echo "<h1>hi</h1>" > payload.html
+    ```
+
+1. Prepare a certificate and private key pair to use for signing the exchange. As of July 2018, we need to use self-signed certificate for testing, since there are no CA that issues certificate with ["CanSignHttpExchanges" extension](https://wicg.github.io/webpackage/draft-yasskin-http-origin-signed-responses.html#cross-origin-cert-req). To generate a signed exchange compatible self-signed key pair with OpenSSL, invoke:
+    ```
+    # Generate prime256v1 ecdsa private key.
+    openssl ecparam -out priv.key -name prime256v1 -genkey
+    # Create a certificate signing request for the private key.
+    openssl req -new -sha256 -key priv.key -out cert.csr \
+      -subj '/CN=example.org/O=Test/C=US'
+    # Self-sign the certificate with "CanSignHttpExchanges" extension.
+    openssl x509 -req -days 360 -in cert.csr -signkey priv.key -out cert.pem \
+      -extfile <(echo "1.3.6.1.4.1.11129.2.1.22 = ASN1:NULL\nsubjectAltName=DNS:example.org")
+    ```
+
+1. Convert the PEM certificate to `application/cert-chain+cbor` format using `gen-certurl` tool.
+    ```
+    # Fill in dummy data for OCSP/SCT, since the certificate is self-signed.
+    gen-certurl -pem cert.pem -ocsp <(echo ocsp) -sct <(echo sct) > cert.cbor
+    ```
+
+1. Host the `application/cert-chain+cbor` created in Step 3 on the HTTPS server. Configure the resource to be served with `Content-Type: application/cert-chain+cbor` HTTP header. The steps below assume the `cert.cbor` is hosted at `https://yourcdn.example.net/cert.cbor`, so substitute the URL to the actual URL in below steps.
+    - Note: If you are using [Firebase Hosting](https://firebase.google.com/docs/hosting/) as your HTTPS server, see an example config [here](https://github.com/WICG/webpackage/blob/master/examples/firebase.json).
+
+1. Generate the signed exchange using `gen-signedexchange` tool.
+    ```
+    gen-signedexchange \
+      -uri https://example.org/hello.html \
+      -content ./payload.html \
+      -certificate cert.pem \
+      -privateKey priv.key \
+      -certUrl https://yourcdn.example.net/cert.cbor \
+      -validityUrl https://example.org/resource.validity.msg \
+      -o example.org.hello.sxg
+    ```
+
+1. Host the signed exchange file `example.org.hello.sxg` on the HTTPS server. Configure the resource to be served with `Content-Type: application/signed-exchange;v=b1` HTTP header.
+    - Note: If you are using [Firebase Hosting](https://firebase.google.com/docs/hosting/) as your HTTPS server, see an example config [here](https://github.com/WICG/webpackage/blob/master/examples/firebase.json).
+
+1. Navigate to the signed exchange URL using a web browser supporting signed exchange.
+    - As of July 2018, you can use Chrome [Dev](https://www.google.com/chrome/?extra=devchannel)/[Canary](https://www.google.com/chrome/browser/canary.html) versions with a command-line flag to enable signed exchange support.
```

Comment:
Might be useful to note the version numbers too?

Reply:
Done